include/dali/core/mm/cuda_vm_resource.h

```diff
@@ -423,8 +423,12 @@ class cuda_vm_resource_base : public memory_resource<memory_kind::device> {
       } catch (const CUDAError &) {
```

Comment:
Is this the dreaded flow by exceptions?
Shouldn't we get a `TryReserve` in this place or something?

(I know this is bit out of scope).

Reply:
(sort of) - this exception doesn't happen frequently and so far hasn't happened even once in any other scenario than requesting an impossibly large block.